packages/visual-editing/src/controller.ts

```diff
@@ -127,30 +129,56 @@ export function createOverlayController({
     })
   }
 
-  function setOverlayCursor() {
+  function setOverlayCursor(element: ElementNode) {
+    // Don't set the cursor if mutations are unavailable
     if (!inFrame || !optimisticActorReady) return
 
-    const hoveredElement = getHoveredElement()
-
-    if (!hoveredElement) return
-
-    const targetSanityData = elementsMap.get(hoveredElement)?.sanity
+    let cursor = undefined
+    // Loop through the hover stack in reverse order to find the element closest
+    // to the top of the stack that has a drag group. Note that this loop finds
+    // the cursor type to set (if any), not the element to set the cursor for.
+    for (const el of hoverStack.toReversed()) {
+      // If the element doesn't have relevant sanity data, continue
+      const targetSanityData = elementsMap.get(el)?.sanity
+      if (!targetSanityData || !isSanityNode(targetSanityData)) continue
+
+      // Resolve the drag group for the currently looped over stack element
+      const dragGroup = resolveDragAndDropGroup(el, targetSanityData, elementSet, elementsMap)
+
+      if (dragGroup) {
+        // If the element in the stack is the currently hovered element, set the
+        // cursor to 'move'. Otherwise the element is a child ancestor, so set
+        // it to 'auto' to override the parent's move cursor
+        const isHoveredElement = element === el
+        cursor = isHoveredElement ? 'move' : 'auto'
```

Comment:
I'm wondering — do we need to modify the children's cursor values? I feel like if we don't need to show the drag handle, we can just do nothing?

Reply:
Huh, this is a good point actually... This workaround of checking the hover stack is necessary because our deferred leave pattern means the leave handler isn't triggered until the element is removed from the hover stack, so the parent element maintains its cursor style and the child elements need to use `auto` to override the cascade.

It might be possible to just not set the cursor style in the deferred leave handler, but in the mouseleave event itself. That would simplify things.